src/spyglass/decoding/v1/sorted_spikes.py

```diff
@@ -479,19 +397,27 @@ def spike_times_sorted_by_place_field_peak(self, time_slice=None):
                 for neuron_ind in neuron_sort_ind
             ]
 
-    def get_ahead_behind_distance(self):
-        # TODO: allow specification of specific time interval
-        # TODO: allow specification of track graph
-        # TODO: Handle decode intervals, store in table
+    def get_ahead_behind_distance(self, track_graph=None, time_slice=None):
+        # TODO: store in table
 
-        classifier = self.load_model()
-        results = self.load_results()
-        posterior = results.acausal_posterior.unstack("state_bins").sum("state")
+        if time_slice is None:
+            time_slice = slice(-np.inf, np.inf)
+
+        classifier = self.fetch_model()
+        posterior = (
+            self.fetch_results()
+            .acausal_posterior.sel(time=time_slice)
+            .unstack("state_bins")
+            .sum("state")
+        )
```

Comment:
Does this need to be squeezed when not estimating transition probabilities?

Reply:
Yes if there is only one interval.